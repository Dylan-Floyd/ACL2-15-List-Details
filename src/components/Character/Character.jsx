import React from 'react'
import styles from './Character.module.css'

export default function Character({ characterData }) {
  const {
    name,
    origin,
    status,
    image
  } = characterData

  return (
    <div className={styles.charCard}>
      <h3 className={styles.name}>{name}</h3>
      <div className={styles.origin}>Origin: {origin}</div>
      <div>Status: {status}</div>
      <img src={image} alt={name} />
    </div>
  )
}
