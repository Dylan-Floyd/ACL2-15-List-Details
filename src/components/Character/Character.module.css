.charCard {
  display: flex;
  flex-direction: column;
  justify-items: flex-start;
  align-items: center;
  width: 250px;
  background-color: #ffffff;
  color:rgb(0, 0, 0);
  padding: .5rem;
  margin: .5rem;
  border-radius: 15px;
  border: 3px solid rgb(114, 48, 28);
  box-shadow: 3px 3px 7px 5px #a9d3e9bb;
  text-align: center;
}

.charCard > img {
  width: 200px;
  margin: .5rem;
  border-radius: 10px;
  border: 2px solid #a9d3e9;
}

.name {
  min-height: 3em;
  text-align: center;
}

.origin {
  min-height: 3em;
}