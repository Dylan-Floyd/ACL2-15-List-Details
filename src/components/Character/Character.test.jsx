import { render } from '@testing-library/react'
import Character from './Character.jsx'

it('should render a character with the provided data', () => {
  const characterData = {
    name: 'Rick Sanchez',
    status: 'Alive',
    image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg'
  }

  const { container } = render(<Character characterData={characterData} />)

  expect(container).toMatchSnapshot()
})