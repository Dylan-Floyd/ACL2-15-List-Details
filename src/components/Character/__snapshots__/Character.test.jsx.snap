// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a character with the provided data 1`] = `
<div>
  <div
    class="charCard"
  >
    <h3
      class="name"
    >
      Rick Sanchez
    </h3>
    <div
      class="origin"
    >
      Origin: 
    </div>
    <div>
      Status: 
      Alive
    </div>
    <img
      alt="Rick Sanchez"
      src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
    />
  </div>
</div>
`;
